datafusion/optimizer/src/utils.rs

```diff
@@ -96,30 +96,173 @@ fn split_conjunction_impl<'a>(expr: &'a Expr, mut exprs: Vec<&'a Expr>) -> Vec<&
 /// ];
 ///
 /// // use split_conjunction_owned to split them
-/// assert_eq!(split_conjunction_owned(expr), split);
+/// assert_eq!(split_conjunction_owned(expr, Operator::And), split);
 /// ```
-pub fn split_conjunction_owned(expr: Expr) -> Vec<Expr> {
-    split_conjunction_owned_impl(expr, vec![])
+pub fn split_conjunction_owned(expr: Expr, op: Operator) -> Vec<Expr> {
```

Comment:
@alamb I think we could support on split on other types.

Reply:
I agree this is useful -- though I would prefer this function to remain named `split_conjunction` (as the `conjunction` implies `AND` and instead have a new function called `split_operator` or something. Maybe I can make a quick PR